backend/controllers/user.js

```diff
@@ -0,0 +1,65 @@
+var User = require('../models/user.js')
+var crypto = require('crypto')
+
+exports.postUser = function (req, res) {
+  var username = req.body.username
+  var password = req.body.password
+  var email = req.body.email
+
+  var user = User.build({ email: email, username: username, password: password })
+
+  add(user, function (success) {
+    res.json({ message: 'User created!' })
+  },
+  function (err) {
+    res.send(err)
+  })
+}
+
+exports.getUser = function (req, res) {
+  retrieveAll(function (users) {
+    if (users) {
+      res.json(users)
+    } else {
+      res.send(401, 'User not found')
+    }
+  }, function (err) {
+    res.send('User not found')
+  })
+}
+
+var retrieveAll = function (onSuccess, onError) {
+  User.findAll({}, {raw: true}).then(onSuccess).catch(onError)
+}
+
+exports.retrieveById = function (user_id, onSuccess, onError) {
+  User.find({where: {id: user_id}}, {raw: true}).then(onSuccess).catch(onError)
+}
+
+var add = function (user, onSuccess, onError) {
```

Comment:
extra function seems complicated, maybe inline this method or put it in the user model